File: VB/HowToInitializeObjectCreatedViaNewAction.Module/MyController.vb
Imports System
Imports System.ComponentModel
Imports System.Collections.Generic
Imports System.Diagnostics
Imports System.Text

Imports DevExpress.ExpressApp
Imports DevExpress.ExpressApp.Actions
Imports DevExpress.Persistent.Base
Imports DevExpress.Persistent.BaseImpl
Imports DevExpress.ExpressApp.SystemModule

Public Class MyController
	Inherits DevExpress.ExpressApp.ViewController

	Public Sub New()
		MyBase.New()

		'This call is required by the Component Designer.
		InitializeComponent()
		RegisterActions(components) 

	End Sub

    Private Sub MyController_FrameAssigned(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles MyBase.FrameAssigned
        Dim standardController As NewObjectViewController = _
         Frame.GetController(Of NewObjectViewController)()
        AddHandler standardController.ObjectCreated, AddressOf standardController_ObjectCreated
    End Sub
    Private Sub standardController_ObjectCreated(ByVal sender As Object, _
         ByVal e As ObjectCreatedEventArgs)
        CType(e.CreatedObject, Task).StartDate = DateTime.Now
    End Sub
End Class

File: CS/HowToInitializeObjectCreatedViaNewAction.Module/MyController.cs
using System;
using System.ComponentModel;
using System.Collections.Generic;
using System.Diagnostics;
using System.Text;

using DevExpress.ExpressApp;
using DevExpress.ExpressApp.Actions;
using DevExpress.Persistent.Base;
using DevExpress.Persistent.BaseImpl;

using DevExpress.ExpressApp.SystemModule;
namespace HowToInitializeObjectCreatedViaNewAction.Module {
   public partial class MyController : ViewController {
      public MyController() {
         InitializeComponent();
         RegisterActions(components);
      }

      private void MyController_FrameAssigned(object sender, EventArgs e) {
         NewObjectViewController standardController =
         Frame.GetController<NewObjectViewController>();
      standardController.ObjectCreated += new EventHandler<ObjectCreatedEventArgs>(standardController_ObjectCreated);
      }

      void standardController_ObjectCreated(object sender, ObjectCreatedEventArgs e) {
         ((Task)e.CreatedObject).StartDate = DateTime.Now;
      }
   }
} 